### PR TITLE
404 jwl pentmonsoon

### PR DESCRIPTION
**Pull request by lee1043**
monsoon sperber metrics

### CODE REVIEW

**Comment by gleckler1** [comment]
@doutriaux1 I think this is ready to be merged unless you are planning to finesse further...

**Comment by doutriaux1** [comment]
yes some revamp needed

**Comment by doutriaux1** [comment]
@gleckler1 pls point me to some input data I'll prep them and turn them into a test

**Comment by doutriaux1** [comment]
@gleckler1 do you think we can use files there: 
https://github.com/PCMDI/pcmdi_metrics/tree/404_jwl_pentmonsoon/src/python/monsoon_sperber/doc

**Comment by gleckler1** [comment]
@doutriaux1 pick any file in /work/cmip5-test/new/amip/atmos/day/pr and extract 1 year of data.

**Comment by gleckler1** [comment]
@doutriaux1  I'll check out branch to verify.

**Comment by gleckler1** [comment]
@doutriaux1  Those files are time series results, not original data.   I think you want the test to include the start-to-finish calcs, correct.   If so, better start with the data I sent you.   To make a smaller file, you could select a time series from one of the domains, under metrics_sperber...
https://github.com/PCMDI/pcmdi_metrics/blob/404_jwl_pentmonsoon/share/default_regions.py

**Comment by lee1043** [comment]
@gleckler1 @doutriaux1 I will follow up on this after I back.

**Comment by doutriaux1** [comment]
@lee1043 don't worry I should be able to finsih it today

**Comment by gleckler1** [comment]
@lee1043 No worries enjoy vacation!

**Comment by doutriaux1** [comment]
@lee1043 you need to test from command line as well as param files

A LOT of option are expected to be in a param file but cannot be passed from command line. This is dure to the changes from @zshaheen not being in yet. We cannot bring in anything that is in the param file. @zshaheen I will merge this in soon. But this is an example why we need the default to be on. Sorry @zshaheen

File: src/python/monsoon_sperber/scripts/driver_monsoon_sperber.py [hunk 1]
```diff
@@ -90,8 +90,8 @@
 param = P.get_parameter()
 
 # Pre-defined options
-mip = param.mip
-exp = param.exp
+mip = param.MIP
+exp = param.EXP
```
Comment (lee1043) [inline src/python/monsoon_sperber/scripts/driver_monsoon_sperber.py]: @doutriaux1 Could you please explain what this change is for?

**Comment by lee1043** [comment]
@gleckler1 code is revised to run with GPCP OBS dataset that given as part of obs4MIP. @doutriaux1 I think it is almost ready except for the test. All parameter are declared before used.

**Comment by doutriaux1** [comment]
@lee1043 thx! I'll take a look soon (Thu?)

**Comment by lee1043** [comment]
@doutriaux1 thanks for following up. Wondering what kind of failure appear?

**Comment by doutriaux1** [comment]
looks like graphic tests

**Comment by lee1043** [comment]
@doutriaux1 is that the numpy warning message?

**Comment by doutriaux1** [comment]
nope it's a conda related bug: https://401-10345204-gh.circle-artifacts.com/0/tests_html/test_pmp_pcoords.html

**Comment by lee1043** [comment]
@doutriaux1 okay thanks for fixing it!

**Comment by doutriaux1** [comment]
@lee1043 please do not push to this branch! Especially debug print statements! I'm trying to bring it into master!

**Comment by doutriaux1** [comment]
Currently the failure are not directly related to your branch

**Comment by lee1043** [comment]
@doutriaux1 oops..... sorry. okay, no more pushing...

**Pull request merged**